### DISCUSSION TITLE
Release v1.6.0

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#1717] [General]
Folks, this morning (New Zealand time 😊) I am prepping to do a new release. 

Here's the current change list.

```
            v1.6.0
      * Adds ColorPicker control
      * Adds Context Menu to TreeView
      * Adds support for custom colors in TreeView
      * Adds Data field to MenuItem
      * Adds Timeout thread safety
      * Adds better docs for Application.Shutdown ()
      * Fixes #1699 Menu closes on button press and menu hot-key text renders correctly
      * Fixes #1675 ContextMenu MenuItems typo
      * Fixes #1715 UI Catalog about box version
      * Fixes #1712 Intermittent Github Action build failures
      * Fixes remaining wide runes render issues including #1693 Character Map
      * Fixes the draw clip bounds issue visible on ScrollView.
      * Fixes UpdateScreen running twice on Application.Refresh because Driver. Refresh already does.
      * Fixes TextView throwing when WordWrap is true on constructor initialization.
      * Fixes RunTimers not checking for key collisions in list when adding to timeouts
      * Fixes erroneous namespace on LineView to match other views
      * Fixes #1723 - ProgressBar with Percentage has extra spaces on the right
      * Fixes nullref on unclean shutdown
```

Remaining Issues/PRs that should be merged before release:

* #1693 Character Map to still not work right - Requires @migueldeicaza to release new NStack.

Any thoughts or concerns?

**Comment by tig** [comment]
I think we should also wait for @migueldeicaza to merge https://github.com/migueldeicaza/NStack/pull/57 into nstack and release a new version of that before we Release 1.6.0. 

I say this because now that @BDisp has done such an awesome job of addressing the wide rune stuff, it would be a real shame for Character Map to still not work right (see #1693).

**Reply by BDisp** [reply]
> I think we should also wait for @migueldeicaza to merge [migueldeicaza/NStack#57](https://github.com/migueldeicaza/NStack/pull/57) into nstack and release a new version of that before we Release 1.6.0.
> 
> I say this because now that @BDisp has done such an awesome job of addressing the wide rune stuff, it would be a real shame for Character Map to still not work right (see #1693).

I absolutely agree with you @tig. Please @migueldeicaza :-)

**Reply by migueldeicaza** [reply]
Apologies!  I am merging the patch blindly to not be in the way.  I trust your judgment!

**Reply by tig** [reply]
@migueldeicaza do you have an ETA when you can release a new  NStack.Core to nuget?

**Reply by tig** [reply]
> @migueldeicaza do you have an ETA when you can release a new NStack.Core to nuget?

Pinging @migueldeicaza.

**Comment by tig** [comment]
Oh, and we should also ensure we've solved (or at least better mitigated) the build failure issue: #1712

**Reply by BDisp** [reply]
> Oh, and we should also ensure we've solved (or at least better mitigated) the build failure issue: #1712

@tig did you already see my comment https://github.com/migueldeicaza/gui.cs/pull/1714#issuecomment-1126282268?

**Comment by tig** [comment]
Woot! v1.6.0 is ready. 

I'm about to push the v1.6.0 tag which will publish to Nuget.org.

**Comment by tig** [comment]
<img width="323" alt="image" src="https://user-images.githubusercontent.com/585482/170579494-9818e621-f418-4c86-9a38-f497af7d6b1b.png">

**Comment by tig** [comment]
Version 1.6.4, which includes a bunch of minor fixes to 1.6.0 has been Released.
